### PR TITLE
Feature/subscription

### PR DESCRIPTION
**Pull request by jesuspg**
@pratid @jframos

### CODE REVIEW

**Comment by flopezag** [comment]
Need to resolve the coverage problems.

File: dashboard/config/dashboard.yml [hunk 1]
```diff
@@ -8,7 +8,34 @@ logging:
...
+  host: 'localhost'
+  port: 8000
   path: '/NGSI10/queryContext'
+
+
+#---------------------------------------------------------------------
+# idm oauth2 (mandatory)
+#---------------------------------------------------------------------
+
+idm:
```
Comment (flopezag) [inline dashboard/config/dashboard.yml:27]: You should not show this data in a public repository...

Comment (flopezag) [inline dashboard/config/dashboard.yml:27]: NTC

File: dashboard/config/dashboard.yml [hunk 2]
```diff
@@ -9,6 +9,27 @@ logging:
...
+  url: ''
+  callbackURL: ''
+
+
+#---------------------------------------------------------------------
+# mailman-api (mandatory)
+#--------------------------------------------------------------------
+mailman:
+  host: 'localhost'
+  port: 8000
```
Comment (flopezag) [inline dashboard/config/dashboard.yml:39]: It is the same that the CB?

File: dashboard/lib/app.js [hunk 3]
```diff
@@ -54,9 +59,13 @@ app.set('view engine', 'jade');
...
+    {
+        src: __dirname + '/stylus',
+        dest: __dirname + "/public/stylesheets",
+        compile: compile
     }
-));
+))
+;
 
 app.use(session({secret: 'ssshhhhh', title_timestamp: ''}));
```
Comment (flopezag) [inline dashboard/lib/app.js]: Can I assume that the secret is a dummy one?

Comment (jesuspg) [inline dashboard/lib/app.js]: The secret is used to hash the session. Secret is required for use sessions. Probably we should change it periodically, but for now, the development for this value is TBD.

**Comment by flopezag** [comment]
are you include in the .gitignore the files with format .otf .svg etc...?

File: dashboard/lib/routes/common.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+/** @export */
+module.exports.notAuthorized = notAuthorizedRender;
+
+/** @export */
+module.exports.parseRoles = parseRoles;
+
+
+/** @export */
+module.exports.addAuthorized = addAuthorized;
```
Comment (flopezag) [inline dashboard/lib/routes/common.js]: add new line at the end of the file

File: dashboard/lib/app.js [hunk 5]
```diff
@@ -77,14 +86,159 @@ app.use(cookieParser());
...
+    // If auth_token is not stored in a session redirect to IDM
+    if (!req.session.access_token) {
+        var path = oa.getAuthorizeUrl();
+        logger.debug({op: 'app#get signin'}, "idm path: " + path);
+        res.redirect(path);
+        // If auth_token is stored in a session cookie it sends a button to get user info
+    } else {
+
+        oa.get('https://account.lab.fiware.org/user/', req.session.access_token, function (e, response) {
+            logger.debug("userinfo: " + response);
```
Comment (pratid) [inline dashboard/lib/app.js:149]: I'd take this URL from a config file

File: dashboard/lib/routes/unsubscribe.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            responseString += data;
+        });
+        mailman_res.on('end', function () {
+            logger.info("response mailman:" + mailman_res.statusCode + " " + mailman_res.statusMessage);
+
+            res.redirect('/');
+        });
+    });
+    mailmain_req.on('error', function (e) {
+        // TODO: handle error.
```
Comment (flopezag) [inline dashboard/lib/routes/unsubscribe.js]: TODO or not TODO?

File: dashboard/lib/public/javascripts/dashboard.js [hunk 7]
```diff
@@ -9,19 +36,21 @@ function displaySections() {
...
     doc.close();
+    calc_height();
 }
 
 function changeTitle(id) {
     var element = document.getElementById(id);
     var subtitle = element.innerText || element.textContent;
     var n = subtitle.search(".TestSuite");
-    var new_subtitle = "SANITY CHECK STATUS | " + capitalizeFirstLetter(subtitle.substring(48, n)) + subtitle.substring(n + 10);
+    var new_subtitle = capitalizeFirstLetter(subtitle.substring(26, n)) + subtitle.substring(n + 10);
```
Comment (pratid) [inline dashboard/lib/public/javascripts/dashboard.js]: Try not to use these magic numbers about substrings.

File: dashboard/lib/views/index.jade [hunk 8]
```diff
@@ -3,24 +3,19 @@ extends layout
...
-
-          div.box1
+            div#frameContainer(style="height:100%")
+            div.box1#box1
 
              #box2.box2
+                p
                 ul.region_list
                    each region in regions
                       - color = region['status'] == 'OK' ?'green' : region['status'] == 'NOK'?'red':'yellow';
```
Comment (flopezag) [inline dashboard/lib/views/index.jade:14]: now we have 4 states
- green  -> ok
- red -> nok
- white or grey -> N/A
- yellow -> pok

File: dashboard/lib/views/index.jade [hunk 9]
```diff
@@ -3,24 +3,19 @@ extends layout
...
                       - filename= region['node'] + "_results.html";
                       if region['status']!='N/A'
                         li.red_and_green
```
Comment (flopezag) [inline dashboard/lib/views/index.jade:17]: red_green_and_yellow

yellow means that all the tests passed except the ones related to Object Storage.

File: dashboard/lib/views/index.jade [hunk 10]
```diff
@@ -28,21 +23,5 @@ block content
                       else
                         li.yellow
```
Comment (flopezag) [inline dashboard/lib/views/index.jade:24]: yellow -> is now white or light grey.

File: dashboard/lib/views/logged.jade [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  block container
+
+          div#frameContainer(style="height:100%")
+          div.box1#box1
+             #box2.box2
+
+                p &nbsp;
+                ul.region_list
+                   each region in regions
+                      - color = region['status'] == 'OK' ?'green' : region['status'] == 'NOK'?'red':'yellow';
```
Comment (flopezag) [inline dashboard/lib/views/logged.jade:14]: Take into consideration the 4 states...

File: dashboard/lib/views/logged.jade [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+                      - region_name= region['node'];
+                      - refresh_authorized = region['authorized'];
+                      if region['status']!='N/A'
+                        li.red_and_green
+                             a( class='#{color}', href='javascript:loadReport("#{filename}")') #{region['node']}
+                             if role=='superuser' || (role=='admin' && refresh_authorized)
+                                button.btn(onclick="javascript:refresh(this,'#{region['node']}')") Refresh
+                             else
+                                button.btn(disabled) Refresh
+                             if region['subscribed']
```
Comment (flopezag) [inline dashboard/lib/views/logged.jade:25]: Can I assume that subscribe/unsubscribe button is shown only when the user is registered (login) with the IdM?

File: dashboard/test/unit/test_cbroker.js [hunk 13]
```diff
@@ -33,9 +33,6 @@ suite('cbroker', function () {
     test('should_return_a_json_with_all_regions_and_status', function () {
         //given
         var json = JSON.parse(fs.readFileSync('test/unit/post1.json', 'utf8'));
-        //when
-        var result = cbroker.parseRegions(json);
-        //then
         var expected = [
             {node: 'Region1', status: 'NOK', timestamp: '2015/05/13 11:10 UTC'},
             {node: 'Region2', status: 'OK', timestamp: '2015/05/13 11:10 UTC'},
```
Comment (flopezag) [inline dashboard/test/unit/test_cbroker.js]: Should we add the status POK and NA, this last one if really exist in the CB.

File: dashboard/lib/views/logged.jade (hunk 12 above)
Comment (jesuspg) [inline dashboard/lib/views/logged.jade:25]: yes!

Comment (flopezag) [inline dashboard/lib/views/logged.jade:25]: NTC

File: dashboard/lib/views/index.jade (hunk 8 above)
Comment (jesuspg) [inline dashboard/lib/views/index.jade:14]: this feature goes to another branch

File: dashboard/lib/views/index.jade (hunk 9 above)
Comment (jesuspg) [inline dashboard/lib/views/index.jade:17]: this feature goes to another branch

File: dashboard/lib/views/index.jade (hunk 10 above)
Comment (jesuspg) [inline dashboard/lib/views/index.jade:24]: this feature goes to another branch

File: dashboard/lib/views/logged.jade (hunk 11 above)
Comment (jesuspg) [inline dashboard/lib/views/logged.jade:14]: this feature goes to another branch

File: dashboard/lib/views/index.jade (hunk 8 above)
Comment (flopezag) [inline dashboard/lib/views/index.jade:14]: NTC

File: dashboard/lib/views/index.jade (hunk 9 above)
Comment (flopezag) [inline dashboard/lib/views/index.jade:17]: NTC

File: dashboard/lib/views/index.jade (hunk 10 above)
Comment (flopezag) [inline dashboard/lib/views/index.jade:24]: NTC

File: dashboard/lib/views/logged.jade (hunk 11 above)
Comment (flopezag) [inline dashboard/lib/views/logged.jade:14]: NTC

File: dashboard/config/dashboard.yml (hunk 2 above)
Comment (jesuspg) [inline dashboard/config/dashboard.yml:39]: port and host data is for a dummy program located in test directory

Comment (flopezag) [inline dashboard/config/dashboard.yml:39]: NTC

**Comment by flopezag** [comment]
LGTM

**Pull request merged**